pkg/cloud/services/compute/instance.go

```diff
@@ -113,6 +104,25 @@ func (s *Service) CreateInstance(openStackCluster *infrav1.OpenStackCluster, mac
 	if err != nil {
 		return nil, err
 	}
+	// verify trunk support if trunk is set on at least one Port.
+	for _, net := range nets {
```

Comment:
Here we're iterating over nets, which is the return value of `constructNetworks()`. constructNetworks returns nets generated from *both* Spec.Networks and Spec.Ports.

I believe this means we're now potentially trunking ports generated for both Spec.Networks and Spec.Ports, where previously we were only trunking Spec.Ports.

I'm not clear on the use case difference between these 2 fields, but I feel like this is a significant change which we should make deliberately. Perhaps @iamemilio or @dulek know?

Reply:
I see your point.
What if I do the checking somewhere around https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/1be73eca005306adeafea58474a4c0248326eae3/pkg/cloud/services/networking/port.go#L166-L179

Reply:
I agree. To avoid that, one option is to move the inheriting logic to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/1be73eca005306adeafea58474a4c0248326eae3/pkg/cloud/services/networking/port.go#L166-L179

And, pass the openstackmachine resource to it.

Reply:
I do think the inheriting logic belongs in `constructNetworks()` and we should make a separate PR if we need to address the inconsistency in inheriting the machine's trunk value between ports and networks. This location in `CreateInstance` is a good place to check if we need to make an API call to check for trunk support in OpenStack.